### DISCUSSION TITLE
Allow use of AdsLib on Windows instead of TcAdsDll

### DISCUSSION THREAD

**Discussion by antonfors** [stlehmann/pyads#428] [Ideas]
Hello!

What are your views on allowing pyads on Windows to fall back on or to try to find AdsLib.dll if TcAdsDll.dll is not found? This should allow users to handle routes in the same way on both Linux and Windows, and enable the usage of pyads without installing the proprietary version of the ADS client library from Beckhoff, but rather through the simple bundling of a .DLL-file.

I did a quick attempt by compiling the modified Beckhoff ADS library that is used by this project for Windows, which did require a minor change to the build script, and then by modifying pyads_ex.py to detect and load AdsLib.dll. It seems promising, but I do not have the ability to test the functionality properly right now as I'm not in close proximity to a Beckhoff PLC today.

**Comment by chrisbeardy** [comment]
I can take a look at your branch and see if it works. You don't need an actual PLC to test, if it works with the TwinCat installed on another PC / a Virtual Machine, it *should* work on a PLC. If it was an easy change and easy to maintain, I would certainly not be against the idea, I think the main reason we use the ADS from Beckhoff is it is probably easier and more foolproof, I know there are some issues when using the linux one.
